### DISCUSSION TITLE
Support for Laravel Fragments

### DISCUSSION THREAD

**Discussion by ralphjsmit** [livewire/livewire#5360] [3. Feature ideas]
Laravel just released a new feature that allows to only render certain [Blade-fragments](https://github.com/laravel/framework/pull/44774). It would be nice if Livewire could support this feature by allowing to only fresh certain fragments in the Blade file. 

As an example where this would be useful. Last week I had a huge Livewire component that showed a long list of orders in a webshop. It was basically an index-page with a lot of rows for all the individual orders. Each order row as a Blade-component and not a Livewire component. The reason for this is that if each page were an individual Livewire component and we had to refresh the complete page, we would have a 100+ AJAX requests for refreshing all the individual order rows. 

Sometimes we only want to refresh a single order row. Currently that's not possible, because we have to refresh the complete Blade-page. There can be significant performance improvements if we had the ability to refresh certain fragments. 

A possible implementation would be like this:

```blade
@foreach($orders as $order)
    @fragment("orders-{$order->id}")
       <x-admin.livewire.order-row :order="$order" />
    @endfragment
@endforeach
```

```blade
<button type="button" wire:click="$emit('$refreshFragment', 'orders-{{ $order->id}}')">
    Refresh row
</button>
```

```php
class OrderIndex extends Component
{
    public function renderFragment(string $fragment): string
    {
        $order = /** */;

        return view('livewire.admin.order.index, ['order' => $order])->fragment($fragment);
    }

    // ..
}

**Comment by Yinci** [comment]
I wholeheartedly agree with this. It could definitely increase performance for larger components. We have a project where a large Livewire form is utilized and it's very slow because it has to refresh the whole thing, whereas if we could only load partials it would be much faster. And sadly, it needs to be live, because every question can add or remove other questions or change answer options. We also have a similar component as yours, so being able to refresh just individual rows would be a lot faster.

**Comment by sap471** [comment]
any progress for this feature?
